shopinvader_membership/tests/test_membership_service.py

```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import timedelta
+
+from odoo import fields
+from odoo.addons.shopinvader.tests.common import CommonCase
+
+
+class TestMembershipService(CommonCase):
+    def setUp(self, *args, **kwargs):
+        super(TestMembershipService, self).setUp(*args, **kwargs)
+        self.membership_line_obj = self.env["membership.membership_line"]
+        self.product_obj = self.env["product.product"]
+        self.partner = self.env.ref("shopinvader.partner_1")
+        str_today = fields.Date.today()
+        self.date_today = fields.Date.from_string(str_today)
+        self.next_month = fields.Date.to_string(
+            self.date_today + timedelta(days=30)
+        )
+        self.product = self.product_obj.create(
```

Comment:
Please move all statements but the service init to setUpClass as they will run only once.